### PR TITLE
[0613] Migrate courses course show page

### PR DESCRIPTION
**Pull request by defong**
### Context

Course show and details page

### Changes proposed in this pull request

Ported across 
- the course show page
- the course details page
- Degree component
  -  https://teacher-training-api-pr-2491.london.cloudapps.digital/view_components/degree_row_content_component/view
- GCSE component
  - https://teacher-training-api-pr-2491.london.cloudapps.digital/view_components/gcse_row_content_component/view
- date presentation (ie short/long date)
- money presentation (ie pound sign and thousand denominator )

Also fixed

`DEPRECATION WARNING: This manually engine loading is deprecated and will be removed in v3.0.0. Remove `require "view_component/engine"`. `

### Guidance to review
Best to compare like for like between 

`https://teacher-training-api-pr-2491.london.cloudapps.digital/publish/organisations/{provider_code}/2022/courses/{course_code}`

vs

`https://qa.publish-teacher-training-courses.service.gov.uk/organisations/{provider_code}/2022/courses/{course_code}`

Best to view per commit

Items omitted
- items to be covered in as another piece of work
  - Change/action links
  - course show page status panel 
- Recruitment cycle 2021and previous
  - UCAS GCSE implementations

image

image

### Checklist

- [ ] Make sure all information from the Trello card is in here
- [ ] Attach to Trello card
- [ ] Rebased master
- [ ] Cleaned commit history
- [ ] Tested by running locally

### CODE REVIEW

**Comment by defong** [comment]
[Migrate Courses - Course show page](https://trello.com/c/L1SDp1cf/613-migrate-courses-course-show-page)

**Review by mbocc — APPROVED** [review]
All good! Thank you :)

**Review by darokel — COMMENTED** [review]
Nice work on this, it was a bit of a monster. I left a few comments. Also the text in the degree component preview seem to be missing styles https://teacher-training-api-pr-2491.london.cloudapps.digital/view_components/degree_row_content_component/view/degree_grade_is_not_required not sure if that will have an impact when used?

File: app/controllers/publish_interface/providers/courses_controller.rb [hunk 1]
```diff
@@ -1,15 +1,36 @@
...
+
       def index
         authorize :provider, :index?
 
         courses_by_accrediting_provider
         self_accredited_courses
       end
 
+      def show
+        fetch_course
```
[the remark this answers is not in the document]
Comment (defong) [reply app/controllers/publish_interface/providers/courses_controller.rb:14]: It conflicts with https://www.rubydoc.info/github/drapergem/draper/Draper%2FDecoratesAssigned:decorates_assigned

aka
`decorates_assigned :course`

**Comment by defong** [comment]
> the text in the degree component preview seem to be missing styles https://teacher-training-api-pr-2491.london.cloudapps.digital/view_components/degree_row_content_component/view/degree_grade_is_not_required not sure if that will have an impact when used?

that component itself was originally poorly `componentise` as it needed it parents from `govuk-summary-list` to fulfil it styling

File: spec/decorators/course_decorator_spec.rb [hunk 2]
```diff
@@ -663,32 +666,32 @@
...
-  #     it "returns the correct details under 'changing your basic details'" do
-  #       expect(decorated_course.listing_basic_details).to eq(["outcome",
-  #                                                             "full time or part time",
-  #                                                             "fee or salary",
-  #                                                             "application open date",
-  #                                                             "course start date"])
-  #     end
-  #   end
+      it "returns the correct details under 'changing your basic details'" do
+        expect(decorated_course.listing_basic_details).to eq(["outcome",
```
[the remark this answers is not in the document]
Comment (defong) [reply spec/decorators/course_decorator_spec.rb:674]: which formatting styling we opting for? wanna turn on the linter to enforce it

[the remark this answers is not in the document]
Comment (defong) [reply spec/decorators/course_decorator_spec.rb:674]: So this one?
https://docs.rubocop.org/rubocop/cops_layout.html#enforcedstyle-with_fixed_indentation-2

File: spec/features/publish_interface/viewing_a_course_spec.rb [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+
+require "rails_helper"
+
+feature "Course show" do
+  before do
+    given_i_am_authenticated_as_a_provider_user
+    when_i_visit_the_course_page
+  end
+
+  let(:course) {
```
[the remark this answers is not in the document]
Comment (defong) [reply spec/features/publish_interface/viewing_a_course_spec.rb]: fixed

File: spec/features/publish_interface/viewing_a_course_details_spec.rb [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+
+require "rails_helper"
+
+feature "Course show" do
+  before do
+    given_i_am_authenticated_as_a_provider_user
+    when_i_visit_the_course_details_page
+  end
+
+  let(:course) {
```
[the remark this answers is not in the document]
Comment (defong) [reply spec/features/publish_interface/viewing_a_course_details_spec.rb]: fixed

**Review by darokel — APPROVED** [review]

File: spec/decorators/course_decorator_spec.rb (hunk 2 above)
[the remark this answers is not in the document]
Comment (defong) [reply spec/decorators/course_decorator_spec.rb:674]: carded as https://trello.com/c/tiLn8usZ/689-code-style-linting

**Pull request merged**